src/module/implements/implement.js

```diff
@@ -39,20 +42,19 @@ class Implement {
   async createEffectsOnItem(item) {
     const implement = await fromUuid(item);
 
-    if (this.implementItem) this.deleteEffectsOnItem();
+    if (this.item) this.deleteEffectsOnItem();
 
     const implementRules = implement.system?.rules ?? [];
     for (const rule of this.rules) {
       implementRules.push(rule);
     }
 
     implement.update({ _id: implement._id, "system.rules": implementRules });
-    this.implementItem = implement.uuid;
+    this.itemId = implement.id;
   }
 
   async deleteEffectsOnItem() {
-    const oldImplement = await fromUuid(this.implementItem);
-    const oldImplementObj = oldImplement.toObject();
+    const oldImplementObj = this.item.toObject();
```

Comment:
This bit to remove all the rules, but just the rules, can be done a little more concisely, and in O(n) time instead of O(n²) (assuming a Set implementation with constant time inserts and lookups) with:

```js
if (this.item) {
  const ruleLabels = new Set(this.rules.map(r => r.label));
  const newRules = this.item.system.rules.filter(r => !ruleLabels.has(r.label));
  await this.item.update({_id: this.item._id, "system.rules": newRules});
}
```

Reply:
Much better and more concise. Thanks!